### ISSUE TITLE
[BUG] bf16bf16bf16 Matmul failed

### ISSUE THREAD

**Issue by WangJialei-A** [intel/graph-compiler#263]
https://github.com/intel/graph-compiler/actions/runs/10463928001/job/28976702848?pr=161

Here is a reproducible module
```
module {
  func.func @entry(%arg0: tensor<16x4096xbf16>, %arg1: tensor<4096x4xbf16>) -> tensor<16x4xbf16> attributes {llvm.emit_c_interface} {
    %cst = arith.constant 0.000000e+00 : bf16
    %0 = tensor.empty() : tensor<16x4xbf16>
    %1 = linalg.fill ins(%cst : bf16) outs(%0 : tensor<16x4xbf16>) -> tensor<16x4xbf16>
    %2 = linalg.matmul {cast = #linalg.type_fn<cast_signed>} ins(%arg0, %arg1 : tensor<16x4096xbf16>, tensor<4096x4xbf16>) outs(%1 : tensor<16x4xbf16>) -> tensor<16x4xbf16>
    return %2 : tensor<16x4xbf16>
  }
}


```

**Comment by WangJialei-A** [comment]
Not fixed, a new error raised. https://github.com/intel/graph-compiler/actions/runs/10465811047/job/28981619563?pr=161
```
python3: /home/ghrunner/_work/graph-compiler/graph-compiler/llvm-project/mlir/include/mlir/IR/Operation.h:683: mlir::Region& mlir::Operation::getRegion(unsigned int): Assertion `index < numRegions && "invalid region index"' failed.
```

**Comment by zhczhong** [comment]
The rest issue happens on `LegalizeToF32` and is the same as https://github.com/intel/graph-compiler/issues/253. @crazydemo could you help take a look on it?

**Comment by crazydemo** [comment]
The fix PR has been submitted to `MLIR community`.[PR#104451](https://github.com/llvm/llvm-project/pull/104451)

**Comment by crazydemo** [comment]
> The fix PR has been submitted to `MLIR community`.[PR#104451](https://github.com/llvm/llvm-project/pull/104451)

The upstream PR has been merged. The issue related to `legalizeToF32` and `mathtolibm` should be resolved by switch `llvm version -> b96f18b20c31449ef9a6878b5c2725a7cf65c552`

**Comment by ZhennanQin** [comment]
@crazydemo Please help to change the llvm revision in GC main branch.
